### PR TITLE
Introducing test coverage report - Closes #49

### LINKED ISSUE #49: Proposal: Test coverage report for the project
Currently the project has no test coverage report, this can be a nice to have to maintain and improve the CSV gem.

Pros:
- We can actively monitor what needs to be done on the current tests and be prepared also for the new contributions.
- Analyze new areas of improvements (on test and the code)

Cons:
- None. Not technical at least.

Plan (if accepted):

- [ ] Add [SimpleCov][simplecov] gem as development dependency and configure the unit test according.
- [ ] Optionally, setup a service to have the info visible(as [CodeClimate][codeclimate])

[simplecov]: https://github.com/colszowka/simplecov
[codeclimate]: https://codeclimate.com

### PR DESCRIPTION
**Pull request by esparta**
This is part of #49: Proposal: Test coverage report for the project

We must change the CodeClimate TestID:

https://codeclimate.com/github/ruby/csv -> Repo Settings -> Test Coverage -> Test reporter ID

Please use that ID as a code suggestion, and we can merge using ruby/csv token instead of my personal one (on [.travis.yml.][travis])

[travis]: https://github.com/ruby/csv/pull/50/files#diff-354f30a63fb0907d4ad57269548329e3R11

### CODE REVIEW

**Comment by esparta** [comment]
This is part of #49: Proposal: Test coverage report for the project

We must change the CodeClimate TestID:

https://codeclimate.com/github/ruby/csv -> Repo Settings -> Test Coverage -> Test reporter ID

Please use that ID as a code suggestion, and we can merge using ruby/csv token instead of my personal one.

File: test/lib/test/unit.rb [hunk 1]
```diff
@@ -9,6 +9,9 @@
 require 'test/unit/testcase'
 require 'optparse'
 
+require 'simplecov'
+SimpleCov.start
+
```
Comment (kou) [inline test/lib/test/unit.rb]: Can we enable this code only when we run our tests on Travis CI?

**Comment by kou** [comment]
I've updated test reporter ID.

File: test/lib/test/unit.rb (hunk 1 above)
Comment (esparta) [reply test/lib/test/unit.rb]: Yes, we can use travis ENVs and restrict the execution based on that. I'll push a change for that.

Comment (esparta) [reply test/lib/test/unit.rb]: Added the restriction to CI env.

Comment (kou) [reply test/lib/test/unit.rb]: Thanks.

**Comment by kou** [comment]
I've merged.

**Comment by esparta** [comment]
Thanks @kou & @hsbt !

**Pull request merged**